### PR TITLE
Support ASP.NET Core PackageReference when targeting .NET Core 3.0 or higher

### LINKED ISSUE #2527: KnownFrameworkReferences need to be specific to a target framework version
Right now, we have `KnownFrameworkReference` items that look something like this in Microsoft.NETCoreSdk.BundledVersions.props:

```xml
    <KnownFrameworkReference Include="Microsoft.DesktopUI"
                              RuntimeFrameworkName="Microsoft.DesktopUI.App"
                              DefaultRuntimeFrameworkVersion="3.0.0-alpha-26829-8"
                              LatestRuntimeFrameworkVersion="3.0.0-alpha-26829-8"
                              TargetingPackName="Microsoft.DesktopUI.App"
                              TargetingPackVersion="3.0.0-alpha-26829-8"
                              />

    <KnownFrameworkReference Include="Microsoft.AspNetCore"
                              RuntimeFrameworkName="Microsoft.AspNetCore.App"
                              DefaultRuntimeFrameworkVersion="3.0.0-alpha1-10062"
                              LatestRuntimeFrameworkVersion="3.0.0-alpha1-10062"
                              TargetingPackName="Microsoft.AspNetCore.App"
                              TargetingPackVersion="3.0.0-alpha1-10062"
                              />
```

These are not specific to the version of .NET Core targeted.  So if you were targeting .NET Core 2.x, you could still add a `FrameworkReference` to Microsoft.DesktopUI, and we would try to add the package reference to the targeting pack.

The `KnownFrameworkReference` items should include a `TargetFrameworkVersion` on them to which they apply, and the `ResolveFrameworkReferences` task should filter them to only look at the ones that apply to the current target framework version.  (We might also want a nice error message if you have a FrameworkReference to something that's available in a different version of .NET Core).

### PR DESCRIPTION
**Pull request by dsplaisted**
Fixes dotnet/cli#10124

Also fixes #2527

### CODE REVIEW

**Review by livarcocc — APPROVED** [review]

**Comment by dsplaisted** [comment]
@DamianEdwards @natemcmaster Would you (or others) like to review the experience and message text here?

Damian had commented:

> Can't we support it but make it a warning?

This PR adds the following warning if you have a PackageReference to Microsoft.AspNetCore.App (with any Version, or no Version):

> NETSDK1080: A PackageReference to Microsoft.AspNetCore.App is not necessary when targeting .NET Core 3.0 or higher. If Microsoft.NET.Sdk.Web is used, the shared framework will be referenced automatically. Otherwise, the PackageReference should be replaced with a FrameworkReference.

If you have a PackageReference to Microsoft.AspNetCore.All, you will get the following **error**:

> NETSDK1079: The Microsoft.AspNetCore.All package is not supported when targeting .NET Core 3.0 or higher. A FrameworkReference to Microsoft.AspNetCore.App should be used instead, and will be implicitly included by Microsoft.NET.Sdk.Web.

How does this look?

**Comment by DamianEdwards** [comment]
Looks good.

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.targets [hunk 1]
```diff
@@ -186,6 +228,14 @@ Copyright (c) .NET Foundation. All rights reserved.
     
   </Target>
 
+  <Target Name="FixAspNetReference" AfterTargets="ResolveFrameworkReferences">
+    <ItemGroup>
+      <PackageReference>
+        <Publish Condition="'%(Identity)' == 'Microsoft.AspNetCore.App'">true</Publish>
```
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.targets]: Shouldn't it come out out ResolveFrameworkReferences this way?

**Review by natemcmaster — APPROVED** [review]
LGTM. Thanks for fixing :)

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.targets (hunk 1 above)
Comment (dsplaisted) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.targets]: Turns out it does, so this was unnecessary.  I think this was a leftover change from previous work I was doing.

**Comment by AraHaan** [comment]
What if you wanted to use this with your own framework and wanted to use the KnownFrameworkReference to add a framework to the known ones and do not explicitly set the things added in this PR?

**Pull request merged**